agent/utils.mjs

```diff
@@ -0,0 +1,659 @@
+    tokenERC20SC.abi = JSON.parse( JSON.stringify( tokensMN.joABI.ERC20_abi ) );
+    tokenERC20SC.address = "" + addressOnSChain;
+}
+
+export async function waitForClonedTokenAppearErc721(
+    sc, tokenERC721SC, joAccountSC, tokensMN, strMainnetName
+) {
+    if( "abi" in tokenERC721SC && typeof tokenERC721SC.abi == "object" &&
+        "address" in tokenERC721SC && typeof tokenERC721SC.address == "string"
+    ) {
+        if( log.verboseGet() >= log.verboseReversed().warning ) {
+            log.write( cc.warning( "Skipping automatic " ) + cc.notice( "ERC721" ) +
+                cc.warning( "instantiation discovery, already done before" ) + "\n" );
+        }
+        return;
+    }
+    const addressCallFrom = joAccountSC.address();
+    const addressOnSChain =
+        await waitForClonedTokenToAppear(
+            sc, "erc721", addressCallFrom, 40, tokensMN, strMainnetName );
+    tokenERC721SC.abi = JSON.parse( JSON.stringify( tokensMN.joABI.ERC721_abi ) );
+    tokenERC721SC.address = "" + addressOnSChain;
+}
+
+export async function waitForClonedTokenAppearErc721WithMetadata(
+    sc, tokenERC721SC, joAccountSC, tokensMN, strMainnetName
+) {
+    if( "abi" in tokenERC721SC && typeof tokenERC721SC.abi == "object" &&
+        "address" in tokenERC721SC && typeof tokenERC721SC.address == "string"
+    ) {
+        if( log.verboseGet() >= log.verboseReversed().warning ) {
+            log.write( cc.warning( "Skipping automatic " ) + cc.notice( "ERC721_with_metadata" ) +
+                cc.warning( " instantiation discovery, already done before" ) + "\n" );
+        }
+        return;
+    }
+    const addressCallFrom = joAccountSC.address();
+    const addressOnSChain =
+        await waitForClonedTokenToAppear(
+            sc, "erc721_with_metadata", addressCallFrom, 40, tokensMN, strMainnetName );
+    tokenERC721SC.abi = JSON.parse( JSON.stringify( tokensMN.joABI.ERC721_with_metadata_abi ) );
+    tokenERC721SC.address = "" + addressOnSChain;
+}
+
+export async function waitForClonedTokenAppearErc1155(
+    sc, tokenERC1155SC, joAccountSC, tokensMN, strMainnetName
+) {
+    if( "abi" in tokenERC1155SC && typeof tokenERC1155SC.abi == "object" &&
+        "address" in tokenERC1155SC && typeof tokenERC1155SC.address == "string"
+    ) {
+        if( log.verboseGet() >= log.verboseReversed().warning ) {
+            log.write( cc.warning( "Skipping automatic " ) + cc.notice( "ERC1155" ) +
+                cc.warning( " instantiation discovery, already done before" ) + "\n" );
+        }
+        return;
+    }
+    const addressCallFrom = joAccountSC.address();
+    const addressOnSChain =
+        await waitForClonedTokenToAppear(
+            sc, "erc1155", addressCallFrom, 40, tokensMN, strMainnetName );
+    tokenERC1155SC.abi = JSON.parse( JSON.stringify( tokensMN.joABI.ERC1155_abi ) );
+    tokenERC1155SC.address = "" + addressOnSChain;
+}
+
+export function encodeUTF8( s ) {
```

Comment:
Maybe we can change to https://www.npmjs.com/package/utf8

Reply:
It's not needed at all. So I will remove it.